sdk/trace/trace_test.go

```diff
@@ -1972,15 +1972,98 @@ func TestWithIDGenerator(t *testing.T) {
 
 			gotSpanID, err := strconv.ParseUint(span.SpanContext().SpanID().String(), 16, 64)
 			require.NoError(t, err)
-			assert.Equal(t, uint64(startSpanID+i), gotSpanID)
+			assert.Equal(t, uint64(startSpanID)+uint64(i), gotSpanID)
 
 			gotTraceID, err := strconv.ParseUint(span.SpanContext().TraceID().String(), 16, 64)
 			require.NoError(t, err)
-			assert.Equal(t, uint64(startTraceID+i), gotTraceID)
+			assert.Equal(t, uint64(startTraceID)+uint64(i), gotTraceID)
 		}()
 	}
 }
 
+func TestIDsRoundTrip(t *testing.T) {
+	gen := defaultIDGenerator()
+	n := 1000
+
+	for i := 0; i < n; i++ {
+		traceID, spanID := gen.NewIDs(context.Background())
+		gotTraceID, err := trace.TraceIDFromHex(traceID.String())
```

Comment:
idea: Maybe we should add fuzz tests for `TraceIDFromHex` and `SpanIDFromHex`.

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/7230